enhancements/microshift/gateway-api-support.md

```diff
@@ -0,0 +1,434 @@
+---
+title: gateway-api-support
+authors:
+  - pacevedom
+reviewers:
+  - copejon
+  - dgn
+  - eslutsky
+  - ggiguash
+  - jerpeter1
+  - pmtk
+  - shaneutt
+approvers:
+  - jerpeter1
+api-approvers:
+  - None
+creation-date: 2024-10-14
+last-updated: 2024-10-14
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-1487
+---
+
+# Gateway API support in MicroShift
+
+## Summary
+At the time of this writing MicroShift supports two types of ingresses:
+[OpenShift routes](https://docs.redhat.com/en/documentation/red_hat_build_of_microshift/4.16/html/networking/microshift-configuring-routes)
+and [Kubernetes ingress](https://docs.redhat.com/en/documentation/red_hat_build_of_microshift/4.16/html/networking/microshift-configuring-routes#nw-ingress-creating-a-route-via-an-ingress_microshift-configuring-routes).
+In order to add more routing features a new project was born in upstream:
+[Gateway API](https://gateway-api.sigs.k8s.io/). This is an official kubernetes
+project focused on L7 routing (support for L4 is experimental), posed to be the
+next standard in ingress, load balancer and service mesh APIs.
+
+This enhancement proposes an implementation to support this new API as an
+optional component in MicroShift.
+
+## Motivation
+Gateway API support will provide users with another way of expressing their
+ingress routing capabilities, also aligned with the way Kubernetes foresees
+this functionality.
+
+### User Stories
+* As a MicroShift admin, I want to add different Gateway resources to the
+  cluster for applications to create routes on them.
+* As a MicroShift application developer, I want to add HTTP routes to specific
+  Gateways so that I can express routing capabilities using an upstream API.
+* As a MicroShift application developer, I want to add GRPC routes to specific
+  Gateways so that I can express routing capabilities using an upstream API.
+
+### Goals
+- Full support for core, non-experimental resources, including gateways, HTTP routes and GRPC routes.
+- Full support for multiple Gateway instances.
+- Full support for north/south [use case](https://gateway-api.sigs.k8s.io/concepts/use-cases/#basic-northsouth-use-case).
+- Lightweight implementation, consuming as little resources as possible.
+
+### Non-Goals
+- Automatically removing Gateway API from the cluster upon RPM uninstall
+- Provide support for experimental resources, such as those listed [here](https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-resources).
+- Support different GatewayClasses with the same controller.
+
```

Comment:
You should state whether service mesh itself is to be a goal/non-goal on MicroShift.  Users will be confused as to why they install OSSM components and cannot use service mesh functionality.

Reply:
I don't see a change corresponding to this....